services/QuillLMS/app/controllers/teacher_infos_controller.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+class TeacherInfosController < ApplicationController
+
+  def create
+    teacher_info = TeacherInfo.create!(
+      user: current_user,
+      minimum_grade_level: minimum_grade_level,
+      maximum_grade_level: maximum_grade_level
+    )
+
+    subject_areas = SubjectArea.where(id: subject_area_ids)
+    teacher_info.subject_areas.push(subject_areas)
```

Comment:
I forgot to add the note about `push` here.  Your analogous implementation in `update` looks good.